src/Sparrow.Server/Utils/ReadOnlyMemoryStream.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Sparrow.Server.Utils;
+
+public class ReadOnlyMemoryStream<T> : Stream
```

Comment:
Pretty sure that we don't actually need that.

Reply:
This allows us to store embeddings from a service without extra allocations or the need to create additional code in the attachment storage.